backend/src/test/java/capstone/backend/controller/OrderToSupplierControllerTest.java

```diff
@@ -183,4 +184,72 @@ void createOrderReturnsNotAcceptableWhenSupplierDoesNotCarryProduct(){
         //THEN
         assertThat(response.getStatusCode(), is(HttpStatus.NOT_ACCEPTABLE));
     }
+
+    @Test
+    void receiveOrder() {
+        //GIVEN
+        Supplier sampleSupplier = supplierRepo.save(sampleSupplier());
+        Product product = productRepo.save(sampleProduct().withSuppliers(List.of(sampleSupplier)));
+        OrderItem orderItem = orderItemRepo.save(sampleOrderItem().withProduct(product));
+        OrderToSupplier pendingOrder = orderRepo.save(new OrderToSupplier(1L, List.of(orderItem), PENDING, sampleSupplier));
+        OrderToSupplierDTO orderToReceive = mapOrder(pendingOrder);
+        OrderToSupplierDTO receivedOrder = mapOrder(pendingOrder);
+        product.setAmountInStock(1);
+        receivedOrder.setStatus(RECEIVED);
+        receivedOrder.setOrderItems(List.of(mapOrderItem(orderItem.withProduct(product))));
+        HttpHeaders headers = utils.createHeadersWithJwtAuth();
+        String URL = BASEURL + String.format("/?id=%d&status=RECEIVED", pendingOrder.getId() );
+        //WHEN
+        ResponseEntity<OrderToSupplierDTO> response = restTemplate.exchange(URL, HttpMethod.PUT, new HttpEntity<>(orderToReceive, headers), OrderToSupplierDTO.class);
+        //THEN
+        assertThat(response.getStatusCode(), is(HttpStatus.OK));
+        assertThat(response.getBody(), is(receivedOrder));
+        assertThat(productRepo.findById(product.getId()).get().getAmountInStock(), is(receivedOrder.getOrderItems().get(0).getQuantity()));
+    }
```

Comment:
Again, you could add the extra check via GET request

Reply:
same here